### PR TITLE
fix undefined values

### PR DESCRIPTION
**Pull request by juliangruber**
using a `undef` singleton, to distinguish between a not found value and a value that you _want_ to be `undefined`

### CODE REVIEW

**Comment by calvinmetcalf** [comment]
How does leveldown handle it, coercing it to a null right?

**Comment by juliangruber** [comment]
if i understand correctly it converts it to an empty slice: https://github.com/Level/leveldown/blob/b5e7afbe6e9ca333ddbfed67a0ef05e291585861/src/leveldown.h#L47

**Comment by juliangruber** [comment]
ah, shall we convert to an empty string? that should be way simpler

**Comment by juliangruber** [comment]
ok, this is way better

File: memdown.js [hunk 1]
```diff
@@ -136,6 +136,7 @@ MemDOWN.prototype._open = function (options, callback) {
 }
 
 MemDOWN.prototype._put = function (key, value, options, callback) {
+  if (typeof value === 'undefined' || value === null) value = ''
```
Comment (calvinmetcalf) [inline memdown.js]: what about a `false` value? aka could this be generalized to `value = value || ''`

Comment (mhart) [inline memdown.js]: Certainly could be generalised to `if (value == null)` (if you didn't want other falsy values)

Comment (juliangruber) [inline memdown.js]: if a value is put as `false` it is expected to be reat as `"false"`, see https://github.com/Level/abstract-leveldown/blob/master/abstract/put-get-del-test.js#L134 (which compares `"false"` to `(false).toString()`)

Comment (juliangruber) [inline memdown.js]: the valid falsy values are

- false
- 0
- NaN

**Comment by calvinmetcalf** [comment]
anyway :+1: feel free to merge

**Comment by juliangruber** [comment]
published as 1.0.1

**Comment by juliangruber** [comment]
oops i don't have publish rights, only @rvagg and @nolanlawson

**Pull request merged**